src/dmd/dsymbolsem.d

```diff
@@ -1148,8 +1148,12 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
             {
                 AliasDeclaration ad = imp.aliasdecls[i];
                 //printf("\tImport %s alias %s = %s, scope = %p\n", toPrettyChars(), aliases[i].toChars(), names[i].toChars(), ad._scope);
-                if (imp.mod.search(imp.loc, imp.names[i]))
+                Dsymbol importedSymbol = imp.mod.search(imp.loc, imp.names[i]);
+                if (importedSymbol)
                 {
+                    import dmd.access : symbolIsVisible;
+                    if (!symbolIsVisible(sc, importedSymbol))
+                         imp.mod.deprecation(imp.loc, "member '%s' is not accessible", imp.names[i].toChars());
```

Comment:
Please add the following comments to make it easier to follow up on changing it to an error.

```D
// @@@DEPRECATED_2018-01@@@
// Deprecation for 1 year after which this should be changed to an error
```

Reply:
Shouldn't this be 2019?

Reply:
It was deprecated in 2018-01. It will be changed to an error in 2019-01.  I read the date as the date is was deprecated, not the date it should be changed to an error.

Reply:
That's not how we use this pattern at Phobos. The string is intended to be greppable in the future, i.e. you can do `grep "@@@DEPRECATED_2019"` for everything that needs maintenance action in 2019 and `grep "@@@DEPRECATED_2019_01"` for everything within a specific month.

Reply:
That doesn't seem right to me, but I'll play along.  It probably needs to be disambiguated somehow.  Please offer @RazvanN7 a suggestion so we can get this merged.

Reply:
Well it's the existing practice to document deprecations, see e.g. https://github.com/wilzbach/phobos/blob/complex/std/datetime/timezone.d#L145

How else are you going to find them? I already did -> Just change to 2018 -> 2019